experimental/dds/tree2/src/shared-tree/sharedTreeChangeTypes.ts

```diff
@@ -0,0 +1,21 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ModularChangeset, SchemaChange } from "../feature-libraries/";
+
+/**
+ * The change format for the SharedTree.
+ * Supports both data and schema changes which can be interleaved as a result of composition.
+ */
+export interface SharedTreeChange {
+	/**
+	 * The changes to apply.
+	 * @remarks while not expressable in TypeScript, these changes should never have two `ModularChangeset`s adjacent in the list.
```

Comment:
It would be nice to document why this restriction exists. It sounds like an optimization: is there a reason somone can't have two changes in a row in a SharedTreeChange, for example as an intermediate version before coalescing them or as part of a deoptimized reference version?

Also technically typescript likely can express that constraint. I wouldn't recommend doing it (its a real mess of recursive types) but I'm think it might be doable.

Reply:
Added a comment.